test/libsolidity/syntaxTests/specialFunctions/abidecode/contract_array.sol

```diff
@@ -0,0 +1,5 @@
+contract C {
+	function f(bytes calldata x) public pure returns (C[] memory c) {
+		c = abi.decode(x, (C[]));
```

Comment:
This needs a semantic test.

Reply:
Added a semantic tests both for ABIEncoderV1 (although there's really not that much to test there) and for ABIEncoderV2 (which correctly reverts with dirty bits).